server/test/support/ResourceCreator.java

```diff
@@ -112,9 +112,17 @@ public Account insertAccount() {
   }
 
   public Applicant insertApplicantWithAccount() {
+    return insertApplicantWithAccount(Optional.empty());
+  }
+
+  public Applicant insertApplicantWithAccount(Optional<String> accountEmail) {
     Applicant applicant = insertApplicant();
     Account account = insertAccount();
 
+    if (accountEmail.isPresent()) {
```

Comment:
not necessary: but I always forget about ifPresent, and that it takes a consumer.  

accountEmail.ifPresent( account::setEmailAddress)

It def reads... different, so just mentioning it if you ever want to use it.

https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html#ifPresent-java.util.function.Consumer-

Reply:
i actually really like this, thanks for showing me!